ui/src/pages/AllStories.tsx

```diff
@@ -21,111 +85,247 @@ function createData(
+      };
+    }
+    case "HANDLE_SWITCH_CHANGE": {
+      const changedVisibilityFoundID = (elem) => elem.ID === action.story.ID;
+      const changedVisibilityContainsID = state.changedVisibility.some(
+        changedVisibilityFoundID
+      );
+
+      if (action.e.target.checked) {
+        return {
+          ...state,
+          visibleState: [...state.visibleState, action.story],
+          visibleTableState: [...state.visibleTableState, action.story],
+          changedVisibility: changedVisibilityContainsID
+            ? state.changedVisibility.filter((e) => e.ID !== action.story.ID)
+            : [...state.changedVisibility, action.story],
+        };
+      } else {
+        return {
+          ...state,
+          visibleState: state.visibleState.filter(
+            (e) => e.ID !== action.story.ID
+          ),
+          visibleTableState: state.visibleTableState.filter(
+            (e) => e.ID !== action.story.ID
+          ),
+          changedVisibility: changedVisibilityContainsID
+            ? state.changedVisibility.filter((e) => e.ID !== action.story.ID)
+            : [...state.changedVisibility, action.story],
+        };
+      }
+    }
+    case "INITIALIZE_AFTER_API": {
+      return {
+        ...state,
+        visibleState: action.rows ? action.rows : [],
+        visibleTableState: action.rows ? action.rows : [],
+        tableData: action.rows ? action.rows : [],
+      };
+    }
+    case "HANDLE_CHECKED_ALL": {
+      return {
+        ...state,
+        selectedRowIds:
+          state.selectedRowIds.length === state.tableData.length
+            ? []
+            : state.tableData.map((story) => story.ID),
+      };
+    }
+    case "HANDLE_CHECKED": {
+      if (action.e.target.checked) {
+        return {
+          ...state,
+          selectedRowIds: [...state.selectedRowIds, action.story.ID],
+        };
+      } else {
+        return {
+          ...state,
+          selectedRowIds: state.selectedRowIds.filter(
+            (e) => e !== action.story.ID
+          ),
+        };
+      }
+    }
+    case "SET_ORDERING": {
+      return {
+        ...state,
+        order: action.order,
+        orderBy: action.orderBy,
+      };
+    }
+    case "SET_TABLE_DATA": {
+      return {
+        ...state,
+        tableData: action.data,
+      };
+    }
+    case "SET_TAB_VALUE": {
+      return {
+        ...state,
+        tabValue: action.newValue,
+      };
+    }
   }
+}
 
-  const [tabValue, setTabValue] = React.useState(0);
-  const [visibleState, setVisibleState] = React.useState([]);
-  const [tableData, setTableData] = useState(rows);
+export const AllStories: React.FC = () => {
+  const [state, dispatch] = useReducer(reducer, initialState);
 
-  const [selectedRowIds, setSelectedRowIds] = useState([]);
-  const [order, setOrder] = useState("desc");
-  const [orderBy, setOrderBy] = useState("id");
+  const fetchStories = (url) =>
+    fetch(url)
+      .then((res) => {
```

Comment:
What happens if there is an error? What should the `.catch()` logic here be?

Reply:
the SWR hook is using the fetcher for getting `allStories` which means it has state handling for the data fetched + the error state, right? Do we need another error handling in the fetcher?